python/ray/data/impl/delegating_block_builder.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+from ray.data.block import Block, T, BlockAccessor
+from ray.data.impl.block_builder import BlockBuilder
+from ray.data.impl.simple_block import SimpleBlockBuilder
+from ray.data.impl.arrow_block import ArrowRow, ArrowBlockBuilder
+
+import pyarrow
```

Comment:
This import is breaking some tests since it introduces a hard arrow dependency.

The import should be moved below.

Reply:
I thought we depend on arrow and pandas if users install ray via `pip install ray[data]`, but it's not the case? https://github.com/ray-project/ray/blob/f814c2af89555d01c1c319ef8feb4cf8058d393a/python/setup.py#L196-L201

Reply:
Fixed.